src/lms-ds-loader/tests/test_df_to_db.py

```diff
@@ -308,3 +308,66 @@ def it_soft_deletes_from_production_table(
         assert adapter_mock.soft_delete_from_production.call_args_list == [
             call(Table.SECTION_ASSOCIATION, SOURCE_SYSTEM)
         ]
+
+
+def describe_when_uploading_assignment_submissions() -> None:
+    @pytest.fixture
+    @patch(
+        "edfi_lms_ds_loader.df_to_db.MssqlLmsOperations.insert_new_records_to_production_for_assignment_and_user"
+    )
+    def when_uploading_assignment_submissions(
+        insert_mock,
+    ) -> Tuple[MagicMock, pd.DataFrame, MagicMock]:
+        # Arrange
+        adapter_mock = MagicMock()
+
+        df = pd.DataFrame([{"SourceSystem": SOURCE_SYSTEM}])
+
+        # Act
+        df_to_db.upload_assignment_submissions(adapter_mock, df)
+
+        return adapter_mock, df, insert_mock
+
+    def it_disables_the_natural_key_index(when_uploading_assignment_submissions) -> None:
+        adapter_mock, _, _ = when_uploading_assignment_submissions
+        assert adapter_mock.disable_staging_natural_key_index.call_args_list == [
+            call(Table.ASSIGNMENT_SUBMISSION)
```

Comment:
I hadn't come across `unittest.mock.call` before. Looks handy.

Reply:
Yeah, `call_args_list` seems to be the way to go if you want to be exact about number of calls and parameters of each call.